go-minitwit/main.go

```diff
@@ -132,31 +118,38 @@ func gravatarURL(email string, size int) string {
 	return fmt.Sprintf("http://www.gravatar.com/avatar/%x?d=identicon&s=%d", hash, size)
 }
 
-func formatMessages(messages []map[string]interface{}) []Message {
-	var formattedMessages []Message
+func formatMessages(messages []MessageUser) []MessageUI {
+	var formattedMessages []MessageUI
+
+	/*
+		if reflect.TypeOf(m.Text).Kind() == reflect.String {
+			filteredMsg.Content = m.Text
+		}
+	*/
+
 	for _, m := range messages {
-		var msg Message
+		var msg MessageUI
 		// Use type assertion for int64, then convert to int
-		if id, ok := m["message_id"].(int64); ok {
-			msg.MessageID = int(id)
+		if reflect.TypeOf(m.MessageID).Kind() == reflect.Int {
+			msg.MessageID = int(m.MessageID)
 		}
-		if authorID, ok := m["author_id"].(int64); ok {
-			msg.AuthorID = int(authorID)
+		if reflect.TypeOf(m.AuthorID).Kind() == reflect.Int {
+			msg.AuthorID = int(m.AuthorID)
 		}
-		if userID, ok := m["user_id"].(int64); ok {
-			msg.User.UserID = int(userID)
+		if reflect.TypeOf(m.UserID).Kind() == reflect.Int {
+			msg.User.UserID = int(m.UserID)
 		}
-		if text, ok := m["text"].(string); ok {
-			msg.Text = text
+		if reflect.TypeOf(m.Text).Kind() == reflect.String {
+			msg.Text = m.Text
 		}
-		if userName, ok := m["username"].(string); ok {
-			msg.Username = userName
+		if reflect.TypeOf(m.Username).Kind() == reflect.String {
+			msg.Username = m.Username
 		}
-		if email, ok := m["email"].(string); ok {
-			msg.Email = email
+		if reflect.TypeOf(m.Email).Kind() == reflect.String {
+			msg.Email = m.Email
 		}
-		if pubDate, ok := m["pub_date"].(int64); ok {
-			pubDateTime := time.Unix(pubDate, 0)
+		if reflect.TypeOf(m.PubDate).Kind() == reflect.Int {
+			pubDateTime := time.Unix(int64(m.PubDate), 0)
```

Comment:
I think you can use normal int here

Reply:
No. time.Unix() takes an int64

Reply:
:)))))))))))))

Reply:
I am making big chances so we dont have so many type conversions.
The reason they are there was because of an old bug i think.

Reply:
Will make a new PR

Reply:
you dont need to do a new PR, just push it to the branch and this PR will be updated